src/components/SearchInput.jsx

```diff
@@ -0,0 +1,109 @@
+import { useState } from 'react';
+import Search from './icon/Search';
+
+export default function SearchInput() {
+  const [isInputOpen, setInputOpen] = useState(false);
+  const [suggestions, setSuggestions] = useState([]);
+  const [searchValue, setSearchValue] = useState('');
+
+  const handleInputChange = (event) => {
+    const { value } = event.target;
+    setSearchValue(value);
+
+    if (value.trim() === '') {
+      setSuggestions([]);
+    } else {
+      fetch(
+        `https://api.artic.edu/api/v1/artworks/search?q=${value}&fields=id,artist_display&limit=5`,
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          const filteredSuggestions = data.data.filter((suggestion) =>
+            suggestion.artist_display
+              .toLowerCase()
+              .includes(value.toLowerCase()),
+          );
+          setSuggestions(filteredSuggestions);
+        });
+    }
+  };
+
+  const handleMouseEnter = () => {
+    setInputOpen(true);
+  };
+
+  const handleMouseLeave = () => {
+    setInputOpen(false);
+    setSuggestions([]);
+  };
+
+  return (
+    <div className="relative">
+      <div className="relative items-end">
+        <input
+          type="text"
+          className={`h-10 w-10 rounded-t-2xl p-2 pl-4 transition-all duration-500 ease-in-out focus:outline-none ${
+            isInputOpen ? 'w-28 sm:w-52' : 'opacity-0'
+          } ${suggestions.length <= 0 ? 'rounded-b-2xl' : ''}`}
```

Comment:
Tu devrais pouvoir gérer l'équivalent du `isInputOpen` avec tailwind le [`hover:`](https://tailwindcss.com/docs/hover-focus-and-other-states#hover-focus-and-active), le [`group`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state), et/ou le [`group-hover`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state)